libpod/pod.go

```diff
@@ -28,14 +32,43 @@ func (p *Pod) Name() string {
 	return p.name
 }
 
-// Creates a new pod
-func newPod() (*Pod, error) {
+// Labels returns the pod's labels
+func (p *Pod) Labels() map[string]string {
+	labels := make(map[string]string)
+	for key, value := range p.labels {
+		labels[key] = value
+	}
+
+	return labels
+}
+
+// Creates a new, empty pod
+func newPod(lockDir string) (*Pod, error) {
 	pod := new(Pod)
 	pod.id = stringid.GenerateNonCryptoID()
-	pod.name = pod.id // TODO generate human-readable name here
+	pod.name = namesgenerator.GetRandomName(0)
 
 	pod.containers = make(map[string]*Container)
 
+	// TODO: containers and pods share a locks folder, but not tables in the
+	// database
+	// As the locks are 256-bit pseudorandom integers, collision is unlikely
+	// But it's something work looking into
+
+	// Path our lock file will reside at
+	lockPath := filepath.Join(lockDir, pod.id)
```

Comment:
I don't think it's an issue, but don't know for sure.  My race condition sensors are tingling just a little bit.  Is there a possibility of a race condition here?  I think it's very very small if at all, but just thought I'd throw it up for consumption in case you think we should add a lock in here before getting the lock file set up.